modules/reindex/src/main/java/org/elasticsearch/index/reindex/GetReindexTaskAction.java

```diff
@@ -0,0 +1,341 @@
+                new ParseField(FATAL_FAILURE));
+            PARSER.declareObjectArray(
+                ReindexTaskWrapperBuilder::setFailures, (p, c) -> BulkByScrollResponse.parseFailure(p), new ParseField(FAILURES)
+            );
+            // since the result of BulkByScrollResponse.Status are mixed we also parse that in this
+            BulkByScrollTask.Status.declareFields(PARSER);
+        }
+
+        private final String id;
+        private final String state;
+        private final long startMillis;
+        private final Long endMillis;
+        private final BulkByScrollTask.Status status;
+        private final List<BulkItemResponse.Failure> bulkFailures;
+        private final List<ScrollableHitSource.SearchFailure> searchFailures;
+        private final ElasticsearchException fatalFailure;
+
+        public ReindexTaskWrapper(StreamInput in) throws IOException {
+            this.id = in.readString();
+            this.state = in.readString();
+            this.startMillis = in.readLong();
+            this.endMillis = in.readOptionalLong();
+            this.status = in.readOptionalWriteable(BulkByScrollTask.Status::new);
+            this.bulkFailures = in.readList(BulkItemResponse.Failure::new);
+            this.searchFailures = in.readList(ScrollableHitSource.SearchFailure::new);
+            this.fatalFailure = in.readOptionalWriteable(ElasticsearchException::new);
+        }
+
+        public ReindexTaskWrapper(String id, String state, long startMillis, @Nullable Long endMillis,
+                                  @Nullable BulkByScrollTask.Status status, @Nullable List<BulkItemResponse.Failure> bulkFailures,
+                                  @Nullable List<ScrollableHitSource.SearchFailure> searchFailures,
+                                  @Nullable ElasticsearchException fatalFailure) {
+            this.id = id;
+            this.state = state;
+            this.startMillis = startMillis;
+            this.endMillis = endMillis;
+            this.status = status;
+            this.bulkFailures = bulkFailures;
+            this.searchFailures = searchFailures;
+            this.fatalFailure = fatalFailure;
+        }
+
+        public static ReindexTaskWrapper fromResponse(String id, long startMillis, Long endMillis, BulkByScrollResponse response) {
+            return new ReindexTaskWrapper(id, getState(response, null), startMillis, endMillis, response.getStatus(),
+                response.getBulkFailures(), response.getSearchFailures(), null);
+        }
+
+        public static ReindexTaskWrapper fromException(String id, long startMillis, Long endMillis, ElasticsearchException exception) {
+            return new ReindexTaskWrapper(id, getState(null, exception), startMillis, endMillis, null,
+                null, null, exception);
+        }
+
+        public static ReindexTaskWrapper fromInProgress(String id, long startMillis, Long endMillis) {
```

Comment:
I think `endMillis` is always null here and should not be passed here?

Also, we should end up including the status here too, but tackling this in a follow-up is fine.